pkg/labels/selector.go

```diff
@@ -47,6 +47,18 @@ func Everything() Selector {
 	return LabelSelector{}
 }
 
+type nothingSelector struct{}
+
+func (n nothingSelector) Matches(_ Labels) bool                         { return false }
+func (n nothingSelector) Empty() bool                                   { return false }
+func (n nothingSelector) String() string                                { return "<null>" }
+func (n nothingSelector) Add(_ string, _ Operator, _ []string) Selector { return n }
+
+// Nothing returns a selector that matches no labels
+func Nothing() Selector {
```

Comment:
Certain use cases for selectors require being able to select nothing, e.g. service selectors. This was part of the original spec for selectors. Obviously services don't always want to select nothing so not bothering to select at all doesn't make sense.